Sources/Testing/Running/Runner.Plan.swift

```diff
@@ -193,6 +193,52 @@ extension Runner.Plan {
     synthesizeSuites(in: &graph, sourceLocation: &sourceLocation)
   }
 
+  /// The basic "run" action.
+  private static let _runAction = Action.run(options: .init())
+
+  /// Determine what action to perform for a given test by preparing its traits.
+  ///
+  /// - Parameters:
+  ///   - test: The test whose action will be determined.
+  ///
+  /// - Returns: A tuple containing the action to take for `test` as well as any
+  ///   error that was thrown during trait evaluation. If more than one error
+  ///   was thrown, the first-caught error is returned.
+  private static func _determineAction(for test: Test) async -> (Action, (any Error)?) {
+    // FIXME: Parallelize this work. Calling `prepare(...)` on all traits and
+    // evaluating all test arguments should be safely parallelizable.
+    await withTaskGroup(returning: (Action, (any Error)?).self) { taskGroup in
```

Comment:
It seems like the rationale for hoisting this out into a standalone function is that you want to enclose it in a task group. Is that right? Could we add a comment to that effect to the doc block?